lib/vector-vrl/web-playground/build.rs

```diff
@@ -24,40 +24,65 @@ fn get_git_hash() -> String {
         .args(["rev-parse", "HEAD"])
         .output()
         .expect("Failed to get git HEAD sha");
-    let mut git_hash = String::from_utf8(output.stdout).unwrap();
-    git_hash.truncate(8);
-    git_hash
+    String::from_utf8(output.stdout).unwrap().trim().to_string()
 }
 
-fn write_build_constants(manifest: &Manifest, dest_path: &Path) -> io::Result<()> {
-    let mut output_file = File::create(dest_path)?;
-    output_file.write_all(
-        "// AUTOGENERATED CONSTANTS. SEE BUILD.RS AT REPOSITORY ROOT. DO NOT MODIFY.\n".as_ref(),
-    )?;
-
-    let create_const_statement =
-        |name, value| format!("pub const {}: &str = \"{}\";\n", name, value);
+fn create_const_statement(name: &str, value: &str) -> String {
+    format!("pub const {}: &str = \"{}\";\n", name, value)
+}
 
+fn write_vector_constants(output_file: &mut File) {
     // TODO: For releases, we should use the manifest.package().version().
     // https://github.com/vectordotdev/vector/issues/18425
-    let vector_version_const = create_const_statement("VECTOR_VERSION", get_git_hash());
+    let vector_git_sha = get_git_hash();
+    let vector_version_statement = create_const_statement("VECTOR_VERSION", &vector_git_sha);
+    output_file
+        .write_all(vector_version_statement.as_bytes())
+        .expect("Failed to write Vector version constant");
+
+    let vector_link = format!("https://github.com/vectordotdev/vector/tree/{vector_git_sha}");
+    let vector_link_statement = create_const_statement("VECTOR_LINK", &vector_link);
     output_file
-        .write_all(vector_version_const.as_bytes())
+        .write_all(vector_link_statement.as_bytes())
         .expect("Failed to write Vector version constant");
+}
 
-    let vrl_version = manifest
+fn write_vrl_constants(manifest: &Manifest, output_file: &mut File) {
+    let vrl_dep = manifest
         .dependencies
         .get("vrl")
-        .unwrap()
+        .expect("missing VRL dependency")
         .detail()
-        .unwrap()
-        .version
-        .clone()
-        .unwrap_or_else(|| "FIXME".into());
-    let vrl_version_const = create_const_statement("VRL_VERSION", vrl_version);
+        .expect("expected detail dependency format");
+
+    let (version, link) = match vrl_dep.version.clone() {
+        None => {
+            let repo = vrl_dep
+                .git
+                .clone()
+                .expect("VRL dependency should use 'version' or 'git'");
+            let version = vrl_dep.rev.clone().expect("VRL git revision not specified");
+            (version.clone(), format!("{repo}/tree/{version}"))
+        }
+        Some(v) => (v.clone(), format!("https://crates.io/crates/vrl/{v}")),
```

Comment:
Probably harmless, but there seems to be a lot of extra cloning going on here. Are they all necessary, particularly since they are only used as references below?

Reply:
Fixed in https://github.com/vectordotdev/vector/pull/19119/commits/9f2b20c452f16d03ae79bbe567fa392f7b50331e